Addition of vehicles.txt to GTFS schedule (#636)
google/transit: gtfs/spec/en/reference.md

> |  `vehicle_class` | ID | Optional | Matches the `vehicle_class` in vehicles.txt to identify the type of vehicle that usually operates on this route. Can be used for information on what type of vehicle to expect when real time data is not available and identify where the vehicle is not the expected type when real time data is available. If trips.txt has a `vehicle_class`, that value overrides the value in routes.txt. |

Reviewer: ```suggestion
|  `vehicle_type` | ID | Optional | Matches the `vehicle_type` in vehicles.txt to identify the type of vehicle that usually operates on this route. Can be used for information on what type of vehicle to expect when real time data is not available and identify where the vehicle is not the expected type when real time data is available. If trips.txt has a `vehicle_type`, that value overrides the value in routes.txt. |
```

|  `cars_allowed` | Enum | Optional | Indicates whether cars are allowed. Valid options are:<br><br>`0` or empty - No car information for the trip.<br>`1` - Vehicle being used on this particular trip can accommodate at least one car.<br>`2` - No cars are allowed on this trip. |
| `safe_duration_factor` | Float | **Optional** | Multiplier applied to travel time estimates calculated for on-demand trips.<br><br>See "Calculating on-demand trip time estimates with safe duration fields" section below for guidance on how to use this and the `safe_duration_offset` fields. |
| `safe_duration_offset` | Float | **Optional** | Fixed offset value in seconds applied to travel time estimates calculated for on-demand trips.<br><br>See "Calculating on-demand trip time estimates with safe duration fields" section below for guidance on how to use this and the `safe_duration_factor` fields. |

> |  `vehicle_class` | ID | Optional | Matches the `vehicle_class` in vehicles.txt to identify the type of vehicle that usually operates on this trip. Can be used for information on what type of vehicle to expect when real time data is not available and identify where the vehicle is not the expected type when real time data is available. This value overrides the value in routes.txt. |

Reviewer: ```suggestion
|  `vehicle_type` | ID | Optional | Matches the `vehicle_type` in vehicles.txt to identify the type of vehicle that usually operates on this trip. Can be used for information on what type of vehicle to expect when real time data is not available and identify where the vehicle is not the expected type when real time data is available. This value overrides the value in routes.txt. |
```

#### Calculating on-demand trip time estimates with safe duration fields
Together, `safe_duration_factor` and `safe_duration_offset` allow an estimation of the longest amount of time a rider can expect the on-demand trip to take, for 95% of cases. Data consumers are expected to use `safe_duration_factor` and `safe_duration_offset` to make the following calculation:<br>`SafeTravelDuration (seconds) = safe_duration_factor × DrivingDuration (seconds) + safe_duration_offset (seconds)`<br>where `DrivingDuration` is the time it would take a private car to travel the distance being calculated for the on-demand service, and `SafeTravelDuration` is the longest amount of time a rider can expect the on-demand trip to take.<br><br>

This calculation should only apply to the portion of a trip that is on-demand. If a service is a deviated-fixed service, or if a rider trip includes a transfer from an on-demand to a fixed-route service, the duration of the fixed-route portion of the trip should be calculated according to the `departure_time` and `arrival_time` fields.

> 

Reviewer: Unrelated change.

|  `attribution_url` | URL | Optional | URL of the organization. |
|  `attribution_email` | Email | Optional | Email of the organization. |
|  `attribution_phone` | Phone number | Optional | Phone number of the organization. |

### vehicles.txt

File: **Optional**

Primary key (`vehicle_label`. `agency_id`)

This file describes the capacity, accessibility and features of individual vehicles or vehicle ranges.

|  Field Name | Type | Presence | Description |
|  ------ | ------ | ------ | ------ |

> |  `vehicle_label` | Text | **Required** | A label that is unique to the vehicle and matches the label shown in the GTFS RealTime label value in the VehicleDescriptor or CarriageDetails or the vehicle identifier in other real-time data sources. If `vehicle_label_high` is provided then this value is the low end of the range. |

Reviewer 2: since `vehicle_label` is not guaranteed to be unique within a fleet, as already mentioned in other comments, could we instead introduce a new field here that is guaranteed to be unique within an agency's fleet? That would make reverse-compatibility easier, and allow `label` to still represent a public facing identifier that a rider might see. This field could then be added to realtime for more exact correlation to `vehicles.txt`

Author: I have added vehicle_id and vehicle_id_high as the new primary key. They correspond to the id value of the VehicleDescriptor and CarriageDetails.

Reviewer 2: Te problem with using `vehicle_id` here is that, in general, `vehicle_id` need not be unique across multiple service dates, and `vehicle_id` need not correspond to any characteristics of the vehicle - for example, at MBTA our `vehicle_id` auto increments within a service date, and the same vehicle can have 7 different `vehicle_id` values in the same week. For this reason and reverse compatibility, you might consider introducing a new field that is guaranteed to have those properties.

Author: Look at the MBTA vehicle location feed, it just add a "y" in front of the vehicle id for buses, I assume to help make them unique.

Adding a third identifier seems like over kill.

|  `attribution_url` | URL | Optional | URL of the organization. |
|  `attribution_email` | Email | Optional | Email of the organization. |
|  `attribution_phone` | Phone number | Optional | Phone number of the organization. |

### vehicles.txt

File: **Optional**

Primary key (`vehicle_id`, `agency_id`)

This file describes the capacity, accessibility and features of individual vehicles or vehicle ranges.

|  Field Name | Type | Presence | Description |
|  ------ | ------ | ------ | ------ |

> |  `vehicle_id` | Text | **Required** | An id that is unique to the vehicle and corresponds to the value in the GTFS Realtime `id` value in the `VehicleDescriptor` or `CarriageDetails` or the vehicle identifier in other real-time data sources. If `vehicle_id_high` is provided then this value is the low end of the range. |

Reviewer 3: It's a bit unclear to me what is the goal with CarriageDetails here. There doesn't seem to be a way to specify in the static that a vehicle is going to have multiple carriage, but you can define it in real time? 

Trying to see how we could potentially specify carriage label when a train has numbered/lettered carriage like google does here 

<img width="1206" height="2622" alt="Image" src="https://github.com/user-attachments/assets/fa1df333-1771-4976-87d7-2a608bf80138" />

Author: You mean to be able to provide the expected vehicle_class/vehicle_type for the carriages where real time data is not available?

Reviewer 3: Yes, or even if real time is available, you might want to know in advance.

> |  `vehicle_id_high` | Text | Optional | If a range of vehicles is represented this is the high end of the range. The number of charactors must be the same as `vehicle_id` and the range covers values with the same number of characters and are between `vehicle_id` and `vehicle_id_high` when using an ASCII string comparison. Not all vehicles covered by the range need to actually exist. |

Reviewer 3: What is the expected behaviour when using UTF-8 strings?

|  `vehicle_label` | Text | Recommended | A label that identifies the vehicle and is visiable to users. Corresponds to the label shown in the GTFS Realtime `label` value in the `VehicleDescriptor` or `CarriageDetails`. If `vehicle_label_high` is provided then this value is the low end of the range. |
|  `vehicle_label_high` | Text | Conditionally Required | If a range of vehicles is represented this is the high end of the range for the vehicle label. The number of characters must be the same as `vehicle_label` and the range covers values with the same number of charactors and are between `vehicle_label` and `vehicle_label_high` when using an ASCII string comparison. Required if both `vehicle_label` and `vehicle_id_high` are included. |
|  `agency_id` | Foreign ID referencing `agency.agency_id` | Recommended | Agency for the specified vehicles. |
|  `license_plate` | Text | Optional | The license plate or number plate of the vehicle. Does not apply if a range of vehicles is being used. |

> |  `vehicle_class` | ID | Recommended | Identifies a class of vehicle that usually runs on certain routes. Ment to link with vehicle_class fields in routes.txt and trips.txt. Can be used to provide vehicle information when label isn't available or highlight when the vehicle doesn't match the expected class and what the major differences are. |

Reviewer 4: typo `Ment` --> `Meant`

> |  `vehicle_class` | ID | Recommended | Identifies a class of vehicle that usually runs on certain routes. Ment to link with vehicle_class fields in routes.txt and trips.txt. Can be used to provide vehicle information when label isn't available or highlight when the vehicle doesn't match the expected class and what the major differences are. |

Reviewer: ```suggestion
|  `vehicle_type` | ID | Recommended | Identifies a type of vehicle that usually runs on certain routes. Meant to link with vehicle_type fields in routes.txt and trips.txt. Can be used to provide vehicle information when label isn't available or highlight when the vehicle doesn't match the expected class and what the major differences are. |
```

| `facility_name` | Text | Optional | The name of the facility where the vehicle is normally stored when not in service, such as the bus garage or rail yard. |

> |  `vehicle_type` | Text | Recommended | Should be one of the two character codes in the Vehicle Type column on the Reference Sheet tab in [A-30 Template](https://www.transit.dot.gov/ntd/30-template) from the US Federal Transit Administration or a one or two word description of the type of vehicle. Examples:<br><br>`BU` - Bus<br>`LR` - Light Rail Vehicle<br>`RP` - Commuter Rail Passenger Coach |

Reviewer 5: When downloading the sheet from https://www.transit.dot.gov/ntd/30-template, I cannot find the codes.

In addition, it is better to have a standardized list or to have some values enumerated in the description here. To make it even simpler, it could just either use a list of standardized code OR a text description, not both. If the purpose of the field is to provide more details on what the vehicle is, then better have it as text and not a list of specific values.

|  `year_manufactured` | Year | Optional | The year the vehicle was manufactured. If a range of vehicles is represented, the earliest year should be used. |
|  `fuel_type` | Text | Optional | The type of fuel used to provide energy for propulsion. It should be from the list of Fuel Types column on the Reference Sheet tab in [A-30 Template](https://www.transit.dot.gov/ntd/30-template) from the US Federal Transit Administration. Examples include `Diesel Fuel`, `Hybrid Diesel`, `Electric Propulsion Power` and `Electric Battery`.|
|  `length` | Positive float | Optional | The length of the vehicle in meters. |
|  `articulated_sections` | integer | Optional | The number of articulated sections. If there are no articulations the number should be 1. |
|  `levels` | integer | Optional | The number of levels available to passengers. Only levels directly above or below other levels count, not where the height of the floor changes within the vehicle. |
| `cab` | enum | Optional | Options are:<br><br>`0` - No cab in the vehicle.<br>`1` - In addition to a passenger area the car has a cab for controlling the vehicle or train.  |

> |  `image_url` | URL | Optional | Fully qualified URL of an image of the vehicle. The image should have as aspect ratio between 1:1 and 2:1 and a have a vertical resolution of at least 500. The image should show the front of the vehicle (except for train cars) and the side the passengers baord. The rights to the image should also free commerical reproduction of the image, for example licenced under [CC-BY](https://creativecommons.org/licenses/by/4.0/). Consumers should only download this image one time for each GTFS update and always serve it to end users themselves. End users should never be automatically downloading directly from the URL provided. |

Reviewer 6: Typo here: baord

> |  `image_url` | URL | Optional | Fully qualified URL of an image of the vehicle. The image should have as aspect ratio between 1:1 and 2:1 and a have a vertical resolution of at least 500. The image should show the front of the vehicle (except for train cars) and the side the passengers baord. The rights to the image should also free commerical reproduction of the image, for example licenced under [CC-BY](https://creativecommons.org/licenses/by/4.0/). Consumers should only download this image one time for each GTFS update and always serve it to end users themselves. End users should never be automatically downloading directly from the URL provided. |

Reviewer: ```suggestion
|  `image_url` | URL | Optional | Fully qualified URL of an image of the vehicle. The image should have as aspect ratio between 1:1 and 2:1 and a have a vertical resolution of at least 500. The image should show the front of the vehicle (except for train cars) and the side the passengers board. The rights to the image should also free commerical reproduction of the image, for example licenced under [CC-BY](https://creativecommons.org/licenses/by/4.0/). Consumers should only download this image one time for each GTFS update and always serve it to end users themselves. End users should never be automatically downloading directly from the URL provided. |
```
I think the licensing should be solved via https://github.com/google/transit/pull/607

|  `vehicle_notes` | Text | Optional | Notable aspects about the vehicles that passengers should know about not covered by other fields. |
|  `capacity_seated` | integer | Optional | The number of seats available to passengers, including ones that may become unavailable due to making space for wheelchairs or bikes. |
|  `capacity_full` | integer | Recommended | The number of passengers that is considered a full load for a vehicle. If passenger occupancy is provided, this should be the number of passengers that is considered Full or 100%. |
|  `capacity_wheelchairs` | integer | Recommended | The number of spots available for passingers with wheelchairs. |
|  `capacity_strollers` | integer | Optional | The number of spots available for strollers. Would include spots used by wheelchairs or strollers. |
|  `capacity_bikes` | integer | Recommended | The number of bicycles the vehicle can carry, either on the inside or outside of the vehicle. |
|  `bike_notes` | Text | Optional | A decription of the location, procedures and restriction of bringing a bicycle on the vehicle. |

> |  `capacity_cars` | integer | Optional | The number of standard size cars the vehicle can carry. Primarily for farries. |

Reviewer 3: ```suggestion
|  `capacity_cars` | integer | Optional | The number of standard size cars the vehicle can carry. Primarily for ferries. |
```
